src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -136,12 +138,21 @@ void CARMACloudPlugin::HandleMobilityOperationMessage(tsm3Message &msg, routeabl
 		//Create an event log object for both positive and negative ACK (ackownledgement), and broadcast the event log
 		tmx::messages::TmxEventLogMessage event_log_msg;
 
-		//acknnowledgement: Flag to indicate whether the received geofence was processed successfully by the CAV	
+		//acknnowledgement: Flag to indicate whether the received geofence was processed successfully by the CMV. 1 mapping to acknowledged by CMV
 		std::transform(acknnowledgement_str.begin(), acknnowledgement_str.end(), acknnowledgement_str.begin(), ::tolower );	
-		acknnowledgement_str.find("true") != std::string::npos ? event_log_msg.set_level(IvpLogLevel::IvpLogLevel_info) : event_log_msg.set_level(IvpLogLevel::IvpLogLevel_warn);
+		acknnowledgement_str.find("1") != std::string::npos ? event_log_msg.set_level(IvpLogLevel::IvpLogLevel_info) : event_log_msg.set_level(IvpLogLevel::IvpLogLevel_warn);
 		event_log_msg.set_description(mo_strategy + ": Traffic control id = " + traffic_control_id + ( CMV_id.length() <= 0 ? "":", CMV Id = " + CMV_id )+ ", reason = " + even_log_description);
 		PLOG(logDEBUG) << "event_log_msg " << event_log_msg << std::endl;
 		this->BroadcastMessage<tmx::messages::TmxEventLogMessage>(event_log_msg);	
+
+		//send negative ack to carma-cloud if not receiving any ack from CMV. acknowledgement_status__acknowledged	= 1	
+		if(acknnowledgement_str.find("1") == std::string::npos )
+		{			
+			char xml_str[10000]; 
+			sprintf(xml_str,"<?xml version=\"1.0\" encoding=\"UTF-8\"?><TrafficControlAcknowledgement><reqid>%s</reqid><msgnum>%s</msgnum><cmvid>%s</cmvid><acknowledgement>%d</acknowledgement><description>%s</description></TrafficControlAcknowledgement>",traffic_control_id.c_str(), msgnum.c_str() ,CMV_id.c_str(),acknowledgement_status::acknowledgement_status__rejected,even_log_description);

```

Comment:
security hotspot sonar scan suggests to replace sprintf with snprintf  to avoid buffer overflow. I think it takes the buffer size as an argument.